pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -169,14 +181,49 @@ func (s *Service) RegisterInstanceWithAPIServerELB(i *infrav1.Instance) error {
 	return nil
 }
 
-// GenerateELBName generates a formatted ELB name
-func GenerateELBName(clusterName string, elbName string) string {
-	return fmt.Sprintf("%s-%s", clusterName, elbName)
+// GenerateELBName generates a formatted ELB name via either
+// concatenating the cluster name to the "-apiserver" suffix
+// or computing a hash for clusters with names above 32 characters.
+func GenerateELBName(clusterName string) (string, error) {
+	standardELBName := generateStandardELBName(clusterName)
+	if len(standardELBName) <= 32 {
+		return standardELBName, nil
+	}
+
+	elbName, err := generateHashedELBName(clusterName, "k8s")
```

Comment:
Still need to push `k8s` down into `generateHashedELBName`

Reply:
thx. done.